src/classes/CRLP_Batch_Base.cls

```diff
@@ -94,40 +94,81 @@ public abstract class CRLP_Batch_Base {
-     */
-    public CRLP_Batch_Base() {
-        this.statefulMapOfOppIdsByParentId = new Map<Id, Set<Id>>();
-        this.summaryObjectType = CRLP_RollupBatch_SVC.getSummaryObjectType(this.jobType);
-        this.isScheduledJob = System.isScheduled();
-        this.customizableRollupSettings = UTIL_CustomSettingsFacade.getCustomizableRollupSettings();
-    }
+    public CRLP_Batch_Base(){}
 
     /*******************************************************************************************************
-     * @description Constructs the class - rollup is for a specific Summary SObject
+     * @description Constructs the class - rollup is for a specific RollupType and RollupTypeFilter
      */
     public CRLP_Batch_Base(CRLP_RollupProcessingOptions.RollupType jobType, CRLP_RollupProcessingOptions.RollupTypeFilter jobFilter) {
-        this();
-        this.jobType = jobType;
-        this.jobFilter = jobFilter;
+        initializeJobData(jobType, jobFilter, null);
     }
 
+
     /*******************************************************************************************************
-     * @description Constructs the class - Rollup is for a specific set of Summary Object Id's
+     * @description Constructs the class - Rollup is a specific RollupType and a specific set of Summary Object Id's
      */
     public CRLP_Batch_Base(CRLP_RollupProcessingOptions.RollupType jobType, List<Id> recordIds) {
-        this();
-        this.jobType = jobType;
-        this.recordIds = recordIds;
+        initializeJobData(jobType, null, recordIds);
     }
 
     /*******************************************************************************************************
-     * @description Constructs the class - Rollup is for a specific set of Summary Object Id's
+     * @description Constructs the class - Rollup is for a specific RollupType
      */
     public CRLP_Batch_Base(CRLP_RollupProcessingOptions.RollupType jobType) {
-        this();
+        initializeJobData(jobType, null, null);
+    }
+
+    /**
+     * @description Called during the start() method for the actual batch job to initialize the summary object, detail
+     * object, key field, and if there are payment rollups that need to be included for the given summary object
+     */
+    @TestVisible
+    protected void initializeJobData(
+            CRLP_RollupProcessingOptions.RollupType jobType,
+            CRLP_RollupProcessingOptions.RollupTypeFilter jobFilter,
+            List<Id> recordIds) {
+
         this.jobType = jobType;
+        if (jobFilter != null) {
+            this.jobFilter = jobFilter;
+        }
+        this.recordIds = recordIds;
+
+        this.summaryObjectType = CRLP_RollupBatch_SVC.getSummaryObjectType(this.jobType);
+        this.statefulCacheOfRollupsToBeProcessed = CRLP_Rollup_SEL.getRollupDefinitionsToUse(this.jobType, this.jobFilter);
+        this.isScheduledJob = System.isScheduled();
```

Comment:
@kslewis Have you verified that `System.isScheduled();` still returns true here when the jobs are executed from the scheduled context (since it's being set from a different location)?

Reply:
I'll try to schedule and test in a scratch org but it should still be set to true since it is still called from within the schedulable class execute method.